app/src/main/java/com/example/henzoshimada/feeltrip/EditMoodActivity.java

```diff
@@ -94,6 +103,59 @@ public void onClick(View v) {
                         dateTime.get(Calendar.MONTH), dateTime.get(Calendar.DAY_OF_MONTH)).show();
             }
         });
+
+    }
+
+    private void submitMood() throws DescriptionTooLongException {
+        ElasticSearchController.AddMoodTask addMoodTask = new ElasticSearchController.AddMoodTask();
+        Mood mood = new Mood("user"); //TODO: pass logged in user to here
+        if(showPublicOn){
+            mood.setPublic();
+        }
+        else {
+            mood.setPrivate();
+        }
+        if(locationOn) {
+            //TODO: mood.setMapPosition();
+            // this is the setter for latitude and longitude
+        }
+        mood.setEmotionalState(emotionalState);
+        mood.setSocialSit(socialSit);
+        mood.setDescription(String.valueOf(inputMoodDescription.getText()), " -Feeling " + emotionalState); //TODO: We might not need to send this append part to Elasticsearch, but rather add it to our displayed description after we fetch from Elasticsearch
+        mood.setDate(dateTime.getTime());
+        // TODO: mood.setImage();
+        // this is the setter for image
+        addMoodTask.execute(mood);
+    }
+
+    public int emojiUnicode(String emotion) {
+        switch(emotion) {
+            case "Angry":
+                return angry;
+            case "Confused":
+                return confused;
+            case "Disgusted":
+                return disgusted;
+            case "Fearful":
+                return fearful;
+            case "Happy":
+                return happy;
+            case "Sad":
+                return sad;
+            case "Shameful":
+                return shameful;
+            case "Cool":
+                return cool;
+            default:
+                return somethingwentwrong; // the secret "only accessible by hacks" face
+        }
+    }
+
+    public String getEmojiByUnicode(int unicode){
+        if (unicode == 0) {
+            return "";
+        }
+        return new String(Character.toChars(unicode));
```

Comment:
Functionality for the unicode emojis will still be here, but this implementation will be changed to accept image Strings instead of unicode ints.